app/controllers/campaigns_controller.rb

```diff
@@ -7,12 +7,13 @@ def display_skip_to_main_navigation?
     false
   end
 
+  ::CampaignPage::Campaign::WHITELIST.each do |campaign_page|
+    method_name = campaign_page.underscore
+    class_eval "def #{method_name}; end"
```

Comment:
Hi @tomas-stefano the reason for creating dynamic methods for all campaign paths are as follows:
  - these dynamic methods are non-RESTful actions for rendering the relevant template of the same name. For example `/en/campaigns/csa` will render `app/views/campaigns/csa.html.erb`.
  - the alternative implementation was to direct all campaign pages through the show action and apply logic on the params[:id]
  - the reason stated above was the security risk flagged by _Hakiri_